griptape/drivers/assistant/griptape_cloud_assistant_driver.py

```diff
@@ -41,8 +43,45 @@ class GriptapeCloudAssistantDriver(BaseAssistantDriver):
     stream: bool = field(default=False, kw_only=True)
     poll_interval: int = field(default=1, kw_only=True)
     max_attempts: int = field(default=20, kw_only=True)
+    auto_create_thread: bool = field(default=True, kw_only=True)
 
     def try_run(self, *args: BaseArtifact) -> BaseArtifact | InfoArtifact:
+        self._init_thread()
+        assistant_run_id = self._create_run(*args)
+        run_result = self._get_run_result(assistant_run_id)
+
+        run_result.meta.update(
+            {"assistant_id": self.assistant_id, "assistant_run_id": assistant_run_id, "thread_id": self.thread_id}
+        )
+
+        return run_result
+
+    def _init_thread(self) -> None:
+        if self.thread_id is None and self.auto_create_thread:
+            if self.thread_alias is None:
+                self.thread_id = self._create_thread()
+            else:
+                try:
+                    self.thread_id = self._create_thread(self.thread_alias)
+                except requests.HTTPError as e:
+                    if e.response.status_code == 400:
+                        thread = self._find_thread_by_alias(self.thread_alias)
+                        self.thread_id = thread["thread_id"]
+                    else:
+                        raise
```

Comment:
cant you remove the first check?
```suggestion
    def _init_thread(self) -> None:
        if self.thread_id is None and self.auto_create_thread:
            try:
                self.thread_id = self._create_thread(self.thread_alias)
            except requests.HTTPError as e:
                if e.response.status_code == 400:
                    thread = self._find_thread_by_alias(self.thread_alias)
                    self.thread_id = thread["thread_id"]
                else:
                    raise
```
maybe also remove the try/except, remove `raise_for_status` and explicitly check the response code for a 201? [existing logic here](https://github.com/griptape-ai/griptape/blob/main/griptape/drivers/memory/conversation/griptape_cloud_conversation_memory_driver.py)